client/modules/IDE/components/PreviewFrame.jsx

```diff
@@ -350,4 +363,28 @@ PreviewFrame.defaultProps = {
   cmController: {}
 };
 
-export default PreviewFrame;
+const mapStateToProps = state => ({
+  files: state.files,
+  htmlFile: getHTMLFile(state.files),
+  ...state.ide,
```

Comment:
I know that the other components that have recently been updated to use `react-redux` aren't like this, but I think it would be better to pull out the specific parts of the redux state that this component depends on. This is because the component will re-render if any of these change, versus the specific parts of the redux state that the component actually depends on.

The other components that are like this should be fixed too, but that's a matter for another PR 😄

Reply:
Yes, I see, that seemed odd anyway. Ok, ill get on it.